### PR TITLE
Fixed #32747 -- Prevented initialization of unused caches.

### PR DESCRIPTION
**Pull request by felixxm**
Thanks Alexander Ebral for the report.

Regression in 98e05ccde440cc9b768952cc10bc8285f4924e1f.

ticket-32747

### CODE REVIEW

**Comment by felixxm** [comment]
@AlexEbral  Can you confirm that it works for you?

**Comment by AlexEbral** [comment]
@felixxm yes, it works for me, thank you!

File: docs/releases/3.2.4.txt [hunk 1]
```diff
@@ -9,4 +9,5 @@ Django 3.2.4 fixes several bugs in 3.2.3.
 Bugfixes
 ========
 
-* ...
+* Prevented, following a regression in Django 3.2, unnecessary initialization
+  of unused caches (:ticket:`32747`).
```
Comment (pope1ni) [inline docs/releases/3.2.4.txt:13]: No need to break the flow of the sentence?
```suggestion
* Prevented unnecessary initialization of unused caches following a regression
  in Django 3.2 (:ticket:`32747`).
```

**Review by pope1ni — APPROVED** [review]
Straightforward. Looks good to me.

**Comment by felixxm** [comment]
Thanks both for checking :+1:

File: docs/releases/3.2.4.txt (hunk 1 above)
Comment (felixxm) [reply docs/releases/3.2.4.txt]: Updated.

**Comment by apollo13** [comment]
Mhm, while I agree that we do have a regression here I am not sure if we should fix it like this. Looking through the cache backends, it doesn't look like `__init__` is supposed to actually "connect" to the cache. For consistency I'd prefer if `all` actually returned all caches. What if we instead remove `self._createdir()` from the `__init__` of the file based cache? This should remove any I/O issues.

**Comment by felixxm** [comment]
> What if we instead remove `self._createdir()` from the `__init__` of the file based cache? This should remove any I/O issues.

Yes, but I'm not sure about making another behavior change :thinking:

**Comment by apollo13** [comment]
Then what about a new argument `initialized_only=False`or so to all? (Or even a new method?) `.all()` is used in one place to clean up all the caches… For further extension on the `BaseConnectionHandler` I'd prefer anything that doesn't already introduce inconsistencies about what `all` means.

**Comment by felixxm** [comment]
@apollo13 I added the `initialized_only=False` kwarg. Can you take a look?

**Review by apollo13 — APPROVED** [review]
I think I like that one :) Thanks! @AlexEbral you check if that also fixes it for you, or if we missed a `.all` call somehwere?

**Comment by AlexEbral** [comment]
@apollo13  yes, this version also works for me.

**Comment by felixxm** [comment]
@AlexEbral Thanks for checking :+1:

**Pull request merged**